mutation/partition_version.cc

```diff
@@ -31,18 +31,7 @@ static void remove_or_mark_as_unique_owner(partition_version* current, mutation_
     }
 }
 
-partition_version::partition_version(partition_version&& pv) noexcept
-    : anchorless_list_base_hook(std::move(pv))
-    , _backref(pv._backref)
-    , _schema(std::move(pv._schema))
-    , _is_being_upgraded(pv._is_being_upgraded)
-    , _partition(std::move(pv._partition))
-{
-    if (_backref) {
-        _backref->_version = this;
-    }
-    pv._backref = nullptr;
-}
+partition_version::partition_version(partition_version&& pv) noexcept = default;
 
```

Comment:
I am not well versed with this area, will leave review for @tgrabiec . Isn't it a problem that we no longer update _backref of both pv and this? Don't we rely on this elsewhere?

Reply:
If we decide to preserve the ctor (as is), apparently we can do this:

// NOLINTBEGIN
ctor
// NOLINTEND

Reply:
yeah, i just realized that we should update the backref pointer.